python/nncase/__init__.py

```diff
@@ -176,7 +176,7 @@ def dump_range_options(self) -> DumpRangeTensorOptions:
 
     def __process_compile_options(self, compile_options: CompileOptions) -> ClCompileOptions:
         self._target = _nncase.Target(compile_options.target)
-        self._compile_options.dump_flags = _nncase.DumpFlags.Nothing
+        self._compile_options.dump_flags = _nncase.DumpFlags.Nothing if not compile_options.dump_ir else _nncase.DumpFlags(int(_nncase.DumpFlags.Rewrite) | int(_nncase.DumpFlags.EGraphCost) | int(_nncase.DumpFlags.PassIR))
```

Comment:
Add [`py::arithmetic`](https://pybind11.readthedocs.io/en/stable/classes.html#enumerations-and-internal-types) to `py::enum_<nncase_dump_flags_t>(m, "DumpFlags")` to remove `int()` cast to `DumpFlags`.

Reply:
changed.